gsa/src/gmp/models/ticket.js

```diff
@@ -0,0 +1,117 @@
+/* Copyright (C) 2019 Greenbone Networks GmbH
+ *
+ * SPDX-License-Identifier: GPL-2.0-or-later
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+import {parseSeverity, parseDate} from 'gmp/parser';
+
+import {isDefined, isModelElement} from 'gmp/utils/identity';
+import {isEmpty} from 'gmp/utils/string';
+
+import Model from '../model';
+
+
+class Ticket extends Model {
+
+  static entityType = 'ticket';
+
+  parseProperties(elem) {
+    const ret = super.parseProperties(elem);
+
+    if (isDefined(elem.assigned_to) && isDefined(elem.assigned_to.user)) {
+      ret.assignedTo = {user: new Model(elem.assigned_to.user, 'user')};
+    }
+    delete ret.assigned_to;
+
+    if (isModelElement(elem.result)) {
+      ret.result = new Model(elem.result, 'result');
+    }
+    else {
+      delete ret.result;
+    }
+
+    if (isModelElement(elem.report)) {
+      ret.report = new Model(elem.report, 'report');
+    }
+    else {
+      delete ret.report;
+    }
+
+    if (isModelElement(elem.task)) {
+      ret.task = new Model(elem.task, 'task');
```

Comment:
Why not use new Task?
Same for all other instances where you create a new Model.

Reply:
Because it won't create any value here. We only get an id. Besides that using specific models within model classes should be avoided if possible to avoid possible cyclic dependencies.